src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -839,37 +884,39 @@ object OrderedRVD {
         }
       }
 
+      val bufferSize = adjustmentsBuffer.size
       val adjustments = indicesBuilder.result().zipWithIndex.map { case (partitionIndex, index) =>
         assert(sortedKeyInfo(partitionIndex).sortedness >= OrderedRVPartitionInfo.TSORTED)
-        val f: (Iterator[RegionValue]) => Iterator[RegionValue] =
-        // In the first partition, drop elements that should go in the last if necessary
-          if (index == 0)
-            if (adjustmentsBuffer.nonEmpty && pkOrd.equiv(min, sortedKeyInfo(adjustmentsBuffer.last.head.index).max))
-              { it: Iterator[RegionValue] =>
-                val itRegion = Region()
-                val rvb = new RegionValueBuilder(itRegion)
-                rvb.start(typ.pkType)
-                rvb.addAnnotation(typ.pkType, min)
-                val rvMin = RegionValue(itRegion, rvb.end())
-                it.dropWhile { rv =>
-                  typ.pkRowOrd.compare(rvMin, rv) == 0
-                }
-              }
+        (partitionIndex, (typ: OrderedRVDType) => {
+          val f: Iterator[RegionValue] => Iterator[RegionValue] =
+          // In the first partition, drop elements that should go in the last if necessary
```

Comment:
I find this sentence confusing. How does the first partition have a "last" partition?

Reply:
replaced with 
```
// In the first adjusted partition, drop elements that belong in the previous adjusted partition
```